scipy/interpolate/tests/test_rgi.py

```diff
@@ -484,6 +482,71 @@ def test_fill_value(self, method):
                                          method=method, bounds_error=False)
         assert np.isnan(interp([10]))
 
+    @parametrize_rgi_interp_methods
+    def test_nonscalar_values(self, method):
+        # Verify that non-scalar valued values also works
+        points = [(0.0, 0.5, 1.0, 1.5, 2.0, 2.5)] * 2 + [
+            (0.0, 5.0, 10.0, 15.0, 20, 25.0)
+        ] * 2
+
+        rng = np.random.default_rng(1234)
+        values = rng.random((6, 6, 6, 6, 8))
+        sample = rng.random((7, 11, 4))
+
+        interp = RegularGridInterpolator(points, values, method=method,
+                                         bounds_error=False)
+        v = interp(sample)
+        assert_equal(v.shape, (7, 11, 8), err_msg=method)
+
+        vs = []
+        for j in range(8):
+            interp = RegularGridInterpolator(points, values[..., j],
+                                             method=method,
+                                             bounds_error=False)
+            vs.append(interp(sample))
+        v2 = np.array(vs).transpose(1, 2, 0)
+
+        assert_allclose(v, v2, err_msg=method)
+
+    @parametrize_rgi_interp_methods
+    @pytest.mark.parametrize("num_trailing_dims", range(1, 5))
```

Comment:
Oh wow. Might be an overkill, I *think* it'd be enough to only test with `num_trailing_dims = 2` or 3 as a single test. See also below.